public/app/core/services/backend_srv.ts

```diff
@@ -284,11 +290,15 @@ export class BackendSrv {
     return deferred.promise;
   }
 
-  deleteDashboards(dashboardUids) {
+  deleteFoldersAndDashboards(folderUids, dashboardUids) {
     const tasks = [];
 
-    for (let uid of dashboardUids) {
-      tasks.push(this.createTask(this.deleteDashboard.bind(this), true, uid));
+    for (let folderUid of folderUids) {
+      tasks.push(this.createTask(this.deleteFolder.bind(this), true, folderUid));
+    }
+
+    for (let dashboardUid of dashboardUids) {
```

Comment:
Proberbly want to delete dashboards first. What would happen if you delete a folder containing a dashboard?

Reply:
Trying to navigate the code :) But I wonder if it makes more sense to do the build delete server side rather then client side. So just send one request with two lists. one with folderuids and one with dashboarduids.

Reply:
The manage dashboards page actually handles this. If you have a folder selected and press delete it will actually do one request for deleting the folder and all of its dashboards. That's the reason for deleting folders first.

Reply:
Hm ok.
Having this order means that someone has to care about the order. I this case the manage dashboards page. 
In that order, it never matters. 

anyway it's not a big issue. Lets ignore it.